data/src/main/java/com/my/kizzy/data/rpc/RpcImage.kt

```diff
@@ -38,6 +38,12 @@ sealed class RpcImage {
         }
     }
 
+    class AppAsset(val id: String) : RpcImage() {
```

Comment:
Just use `DiscordImage` and save the image links as 

`app-assets/962990036020756480/1300361266212241430.png?size=512`

mp prefix should add the discord cdn

Reply:
Using asset IDs combined with application ID directly gets the asset from Discord CDN
https://discord.com/developers/docs/events/gateway-events#activity-object-activity-asset-image

Reply:
Also app assets aren't image protocols
![image](https://github.com/user-attachments/assets/e521c536-f1cf-4b0e-b83d-bfa6cc6011ec)

Reply:
Yes but there are only 3 assets and creating a whole class for them doesn't seem to be a good idea. Also if we want to support use  of application id provided by users this might break but using the assets url directly won't.

Reply:
> Also if we want to support use of application id provided by users

Only reason for doing this is for the **application name**, which you *still* have to pass in the Activity name as an Activity protocol as its a **required** field.
https://discord.com/developers/docs/events/gateway-events#activity-object

It would be more practical if you allow users to set their own Activity name

> but using the assets url directly won't.

This would allow uses to use their own app assets if you do support it

Reply:
Sending the readily-available assets to the Worker would be more overhead than making a small class in the app tbh. With all due respect it doesn't make sense :/

Reply:
> Only reason for doing this is for the application name, which you still have to pass in the Activity name as an Activity protocol as its a required field.

I think you misunderstood me, what I meant was if this gets merged. And we have an option to set `application_id` in the future, so people using different `application_id` will have problems for setting these images as they are tied to a different application

> This would allow uses to use their own app assets if you do support it

Uploading images from gallery was added for this specific reason so people won't have to go to dev portal and upload their assets

Reply:
> Sending the readily-available assets to the Worker would be much more overhead than making a 5-line class in the app that returns a string as-is. With all due respect it doesn't make sense :/

We are not sending the assets to the worker they will be used by discord directly

See about `mp:` https://discord.com/developers/docs/events/gateway-events#activity-object-activity-asset-image

No too sure but the prefix should directly support all these different types of links
https://discord.com/developers/docs/reference#image-formatting

Reply:
> No too sure but the prefix should directly support all these different types of links [discord.com/developers/docs/reference#image-formatting](https://discord.com/developers/docs/reference#image-formatting)

Surprisingly that does work, despite having an `mp:` prefix meant for Media Proxy

Reply:
That worked? Nice!!.

Thanks for checking that I am actually out for some work

Reply:
Surprisingly yeah, maybe I shouldn't trust the docs that much lol

> Thanks for checking that I am actually out for some work

Yeah np take your time 👌